app/src/main/java/chat/rocket/android/chatrooms/adapter/RoomViewHolder.kt

```diff
@@ -4,80 +4,84 @@ import android.content.res.Resources
 
+class RoomViewHolder(itemView: View, private val listener: (RoomUiModel) -> Unit) :
+    ViewHolder<RoomItemHolder>(itemView) {
     private val resources: Resources = itemView.resources
-    private val channelUnread: Drawable = resources.getDrawable(R.drawable.ic_hashtag_black_12dp)
-    private val channel: Drawable = resources.getDrawable(R.drawable.ic_hashtag_12dp)
-    private val groupUnread: Drawable = resources.getDrawable(R.drawable.ic_lock_black_12_dp)
-    private val group: Drawable = resources.getDrawable(R.drawable.ic_lock_12_dp)
-    private val online: Drawable = resources.getDrawable(R.drawable.ic_status_online_12dp)
-    private val away: Drawable = resources.getDrawable(R.drawable.ic_status_away_12dp)
-    private val busy: Drawable = resources.getDrawable(R.drawable.ic_status_busy_12dp)
-    private val offline: Drawable = resources.getDrawable(R.drawable.ic_status_invisible_12dp)
+    private val channelIcon: Drawable = resources.getDrawable(R.drawable.ic_hashtag_12dp)
+    private val groupIcon: Drawable = resources.getDrawable(R.drawable.ic_lock_12_dp)
+    private val onlineIcon: Drawable = resources.getDrawable(R.drawable.ic_status_online_12dp)
+    private val awayIcon: Drawable = resources.getDrawable(R.drawable.ic_status_away_12dp)
+    private val busyIcon: Drawable = resources.getDrawable(R.drawable.ic_status_busy_12dp)
+    private val offlineIcon: Drawable = resources.getDrawable(R.drawable.ic_status_invisible_12dp)
 
     override fun bindViews(data: RoomItemHolder) {
         val room = data.data
         with(itemView) {
             image_avatar.setImageURI(room.avatar)
             text_chat_name.text = room.name
 
+            if (room.status != null && room.type is RoomType.DirectMessage) {
+                image_chat_icon.setImageDrawable(getStatusDrawable(room.status))
+            } else {
+                image_chat_icon.setImageDrawable(getRoomDrawable(room.type))
+            }
+
             if (room.lastMessage != null) {
-                text_last_message.isVisible = true
                 text_last_message.text = room.lastMessage
+                text_last_message.isVisible = true
             } else {
                 text_last_message.isGone = true
             }
 
             if (room.date != null) {
-                text_last_message_date_time.isVisible = true
-                text_last_message_date_time.text = room.date
+                text_timestamp.text = room.date
+                text_timestamp.isVisible = true
             } else {
-                text_last_message_date_time.isGone = true
+                text_timestamp.isInvisible = true
             }
 
-            if (room.unread != null) {
+            if (room.alert) {
+                if (room.unread == null) text_total_unread_messages.text = "!"
+                if (room.unread != null) text_total_unread_messages.text = room.unread
```

Comment:
this is actually an `else` :)
also you can do `text_total_unread_messages.text = room.unread ?: "!"`